core/src/test/scala/io/aiven/guardian/kafka/Generators.scala

```diff
@@ -37,13 +35,6 @@ object Generators {
     timestampType
   )
 
-  private val keyOffsetMap = new ConcurrentHashMap[String, AtomicLong]().asScala
```

Comment:
Aside from offsets not being based keys, using a counter in this fashion generated incorrect offsets because if a scalacheck generator would fail it would reuse the existing counter rather than resetting it to 0.

In summary when using generators it needs be completely stateless.